src/charm.py

```diff
@@ -143,67 +143,41 @@ def main(self, event):
-                    "roles": [
-                        {
-                            "global": True,
-                            "rules": [
-                                {
-                                    "apiGroups": ["dex.coreos.com"],
-                                    "resources": ["*"],
-                                    "verbs": ["*"],
-                                },
-                                {
-                                    "apiGroups": ["apiextensions.k8s.io"],
-                                    "resources": ["customresourcedefinitions"],
-                                    "verbs": ["create"],
-                                },
-                            ],
-                        },
-                    ],
-                },
-                "containers": [
-                    {
-                        "name": "dex-auth",
-                        "imageDetails": image_details,
-                        "command": ["dex", "serve", "/etc/dex/cfg/config.yaml"],
-                        "ports": [{"name": "http", "containerPort": port}],
-                        "envConfig": {
-                            "CONFIG_HASH": config_hash.hexdigest(),
-                            "KUBERNETES_POD_NAMESPACE": self.model.name,
-                        },
-                        "volumeConfig": [
-                            {
-                                "name": "config",
-                                "mountPath": "/etc/dex/cfg",
-                                "files": [
-                                    {
-                                        "path": "config.yaml",
-                                        "content": config,
-                                    },
-                                ],
-                            },
-                        ],
-                    }
-                ],
-                "kubernetesResources": {
-                    "customResourceDefinitions": [
-                        {"name": crd["metadata"]["name"], "spec": crd["spec"]}
-                        for crd in yaml.safe_load_all(Path("src/crds.yaml").read_text())
-                    ],
-                },
-            }
-        )
-        self.model.unit.status = ActiveStatus()
+        context = {
+            "name": self.model.app.name,
+            "namespace": self.model.name,
+            "port": self.model.config["port"],
+            "config_yaml": config,
+            "config_hash": config_hash.hexdigest(),
+        }
+
+        return [
+            obj
+            for path in glob("src/manifests/*.yaml")
+            for obj in codecs.load_all_yaml(Path(path).read_text(), context=context)
+        ]
+
+    def set_manifest(self, manifest):
+        client = Client()
+        errors = []
+
+        for resource in manifest:
+            try:
+                client.create(resource)
+            except ApiError as err:
+                if err.status == 409:
```

Comment:
is this tested?  I **think** it should be:
```suggestion
                if err.status.code == 409:
```
but I might misunderstand something

Reply:
Nope, not tested, will push up update that fixes that and a few other things